src/utils/p2p.ts

```
@@ -134,3 +101,41 @@ export const getPublicIp = async (): Promise<Result<string, FarcasterError>> =>
     }
   });
 };
+
+/* -------------------------------------------------------------------------- */
+/*                               Private Methods                              */
+/* -------------------------------------------------------------------------- */
+
+const checkIpAddr = (ipAddr: string): HubResult<void> => {
+  const parseListenIpAddrResult = parseAddress(ipAddr);
+  if (parseListenIpAddrResult.isErr()) return err(parseListenIpAddrResult.error);
+
+  // DISCUSS: we previously had a null check here (which seems impossible), but we have no
+  // empty string check which is possible
+  const parsedListenIPAddr = parseListenIpAddrResult.value;
+
+  const toOptionsResult = Result.fromThrowable(
+    () => parsedListenIPAddr.toOptions(),
+    (error) => err(error)
+  )();
+  // An IP address should not have options and should throw if well-formed
+  if (toOptionsResult.isErr()) return ok(undefined);
+
+  const options = toOptionsResult.value;
+  if (options.port !== undefined || options.transport !== undefined) {
+    return err(new HubError('bad_request', 'unexpected multiaddr transport/port information'));
+  }
+  return ok(undefined);
+};
+
+const checkCombinedAddr = (ipAddr: string): HubResult<void> => {
+  return parseAddress(ipAddr).match(
+    (addr) => {
+      // DISCUSS: can toOptions throw here? Should we fail if it does?
```

@sagar-a16z should we be throwing if this throws? or is ignoring the error intentional?

---

We should just catch and return the error.